src/Instrumentation/MySqli/src/MySqliInstrumentation.php

```diff
@@ -0,0 +1,815 @@
+            $span->addLink($spanContext);
+        }
+
+        self::addTransactionLink($tracker, $span, $mysqli);
+    }
+
+    private static function nextResultPostHook(CachedInstrumentation $instrumentation, MySqliTracker $tracker, $obj, array $params, mixed $retVal, ?\Throwable $exception)
+    {
+
+        $mysqli = $obj ? $obj : $params[0];
+
+        $errorStatus = ($retVal === false && !$exception) ? (strlen($mysqli->error) > 0 ? $mysqli->error : null) : null;
+
+        $attributes = $tracker->getMySqliAttributes($mysqli);
+
+        $currentQuery = $tracker->getNextMySqliMultiQuery($mysqli);
+
+        // it was just a call to check if there is a pending query
+        if ($currentQuery === null || ($retVal === false && !$errorStatus && !$exception)) {
+            self::logDebug('nextResultPostHook span dropped', ['exception' => $exception, 'obj' => $obj, 'retVal' => $retVal, 'params' => $params, 'currentQuery' => $currentQuery]);
+            self::dropSpan();
+
+            return;
+        }
+
+        if ($currentQuery) {
+            $attributes[TraceAttributes::DB_STATEMENT] = mb_convert_encoding($currentQuery, 'UTF-8');
+            $attributes[TraceAttributes::DB_OPERATION_NAME] = self::extractQueryCommand($currentQuery);
+        }
+
+        if ($retVal === false || $exception) {
+            //TODO use constant from comment  after sem-conv update
+            $attributes[/*TraceAttributes::DB_RESPONSE_STATUS_CODE*/ 'db.response.status_code'] =  $mysqli->errno;
+        }
+
+        self::endSpan($attributes, $exception, $errorStatus);
+    }
+
+    private static function changeUserPostHook(CachedInstrumentation $instrumentation, MySqliTracker $tracker, $obj, array $params, mixed $retVal, ?\Throwable $exception)
+    {
+
+        if ($retVal != true) {
+            return; //TODO create error span?
+        }
+
+        $mysqli = $obj ? $obj : $params[0];
+
+        $tracker->addMySqliAttribute($mysqli, TraceAttributes::DB_USER, $params[$obj ? 0 : 1]);
+        if (($database = $params[$obj ? 2 : 3] ?? null) !== null) {
+            $tracker->addMySqliAttribute($mysqli, TraceAttributes::DB_NAMESPACE, $database);
+        }
+
+    }
+
+    private static function selectDbPostHook(CachedInstrumentation $instrumentation, MySqliTracker $tracker, $obj, array $params, mixed $retVal, ?\Throwable $exception)
+    {
+
+        if ($retVal != true) {
+            return; //TODO create error span?
+        }
+        $tracker->addMySqliAttribute($obj ? $obj : $params[0], TraceAttributes::DB_NAMESPACE, $params[$obj ? 0 : 1]);
+    }
+
+    private static function preparePostHook(CachedInstrumentation $instrumentation, MySqliTracker $tracker, $obj, array $params, mixed $stmtRetVal, ?\Throwable $exception)
+    {
+
+        if ($exception || !$stmtRetVal instanceof mysqli_stmt) {
+            self::logDebug('mysqli::prepare failed', ['exception' => $exception, 'obj' => $obj, 'retVal' => $stmtRetVal, 'params' => $params]);
```

Comment:
there are a few of these logDebug calls - should they be logged like this, or recorded as an exception on the active span, which [the spec requires](https://opentelemetry.io/docs/specs/otel/trace/exceptions/#recording-an-exception)

Reply:
Thank you. I’ve implemented instrumentation for prepare, as it makes sense because it involves IO with the database.